src/main/java/net/datafaker/service/FakeValuesContext.java

```diff
@@ -58,40 +58,41 @@ public static FakeValuesContext of(Locale locale, String filename, String path,
     }
 
     private static String getFilename(Locale locale) {
-        final String lang = language(locale);
+        String lang = language(locale);
         if ("".equals(locale.getCountry())) {
             return lang;
+        } else {
+            return lang + "-" + locale.getCountry();
         }
-        return lang + "-" + locale.getCountry();
     }
 
     /**
-     * If you new up a locale with "he", it gets converted to "iw" which is old.
+     * If you create a locale with "he", it gets converted to "iw" which is old.
      * This addresses that unfortunate condition.
      */
-    private static String language(Locale l) {
-        switch (l.getLanguage()) {
-            case "iw": return "he";
-            case "in": return "id";
-            case "ji": return "yi";
-        }
-        return l.getLanguage();
+    private static String language(Locale locale) {
+        return switch (locale.getLanguage()) {
+            case "iw" -> "he";
+            case "in" -> "id";
+            case "ji" -> "yi";
+            default -> locale.getLanguage();
+        };
     }
 
     public Locale getLocale() {
-        return sLocale.getLocale();
+        return singletonLocale.getLocale();
     }
 
     public String getFilename() {
         return filename;
     }
 
-    public Set<String> getPath() {
-        return pathes;
+    public Set<String> getPaths() {
```

Comment:
I'm m not against changing this but it is technically a breaking change to the API.

(Renaming public methods others may depend on.)

Reply:
Same with the setter

Reply:
That's a fair comment. Is this part of our public api? I can deprecate this one and introduce the second method, that's no problem.

Reply:
That's safest. Tag the existing one deprecated for removal. It can even then simply call the new method(s) (to avoid duplication).

Reply:
I guess with `FakeValueContext` it is still kind of ok since there is not so many cases 1if not none.
And the cases are very advanced (even not covered in doc =), probably another point to improve)
So if there is any user who were able to use datafaker for such use case, i'm pretty sure that method rename isnot a problem then...

however agree, if we are talking about much stricter approach then may be ...

Reply:
I also think the chance of breaking is probably low. I'll just merge it as is.